core/cog-prefix-routes-handler.c

```diff
@@ -0,0 +1,341 @@
+cog_prefix_routes_handler_mount (CogPrefixRoutesHandler *self,
+                                 const char             *path_prefix,
+                                 CogRequestHandler      *handler)
+{
+    g_return_val_if_fail (COG_IS_PREFIX_ROUTES_HANDLER (self), FALSE);
+    g_return_val_if_fail (path_prefix != NULL, FALSE);
+    g_return_val_if_fail (path_prefix[0] == '/', FALSE);
+    g_return_val_if_fail (COG_IS_REQUEST_HANDLER (handler), FALSE);
+
+    if (g_hash_table_contains (self->routes, path_prefix))
+        return FALSE;
+
+    g_hash_table_insert (self->routes,
+                         g_strdup (path_prefix),
+                         g_object_ref (handler));
+    return TRUE;
+}
+
+/**
+ * cog_prefix_routes_handler_unmount:
+ * @self: a #CogPrefixRoutesHandler
+ * @path_prefix: URI path prefix for a configured route
+ *
+ * Removes a previously configured route.
+ *
+ * Removes a route that matches @path_prefix previously configured
+ * using cog_prefix_routes_handler_mount() or
+ * cog_prefix_routes_handler_mount_path().
+ *
+ * Returns: Whether the route was found and removed.
+ */
+gboolean
+cog_prefix_routes_handler_unmount (CogPrefixRoutesHandler *self,
+                                   const char             *path_prefix)
+{
+    g_return_val_if_fail (COG_IS_PREFIX_ROUTES_HANDLER (self), FALSE);
+    g_return_val_if_fail (path_prefix != NULL, FALSE);
+
+    return g_hash_table_remove (self->routes, path_prefix);
+}
+
+/**
+ * cog_prefix_routes_handler_mount_path:
+ * @self: a #CogPrefixRoutesHandler
+ * @path_prefix: URI path prefix to match
+ * @base_path: Path to a local directory
+ *
+ * Adds a route to the handler pointing to a directory.
+ *
+ * This is a convenience method which convenienceonfigures a route
+ * matching @path_prefix in URI paths, and creates a new
+ * [class@Cog.DirectoryFilesHandler] for @base_path to handle requests
+ * for the route. The @path_prefix must contain a leading
+ * slash character (`/`).
+ *
+ * Returns: Whether the route was successfully added.
+ */
+gboolean
+cog_prefix_routes_handler_mount_path (CogPrefixRoutesHandler *self,
+                                      const char             *path_prefix,
+                                      const char             *base_path)
+{
+    g_return_val_if_fail (COG_IS_PREFIX_ROUTES_HANDLER (self), FALSE);
+    g_return_val_if_fail (path_prefix != NULL, FALSE);
+    g_return_val_if_fail (base_path != NULL, FALSE);
+
+    g_autoptr(GFile) path_file = g_file_new_for_path (base_path);
+    g_return_val_if_fail (cog_directory_files_handler_is_suitable_path (path_file, NULL), FALSE);
+
+    unsigned num_components = 0;
+    for (size_t i = 0; path_prefix[i] != '\0'; i++)
+        if (path_prefix[i] == '/')
+            num_components++;
+
+    g_autoptr(CogRequestHandler) handler =
+        g_object_new (COG_TYPE_DIRECTORY_FILES_HANDLER,
+                      "base-path", path_file,
+                      "strip-components", num_components);
```

Comment:
I have realized that this call to `g_object_new()` is missing the terminating `NULL` at the end of the variable list of arguments, I had a release build crash at me just a moment ago because this :facepalm: 

The updated PR will fix this as well.

Reply:
Good catch.